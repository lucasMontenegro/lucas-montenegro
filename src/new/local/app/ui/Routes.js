import React from "react"
import { BrowserRouter, Switch, Route } from "react-router-dom"
import App from "new/local/app/ui/App"
import Examples from "new/local/paperbase/Examples"
import isProduction from "new/local/utils/isProduction"
export default function Routes () {
  return (
    <BrowserRouter>
      <Switch>
        <Route component={App} />
        {isProduction() || <Route path="/examples" component={Examples} />}
      </Switch>
    </BrowserRouter>
  )
}