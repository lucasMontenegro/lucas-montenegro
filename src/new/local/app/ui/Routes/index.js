import React from "react"
import { BrowserRouter, Switch, Route } from "react-router-dom"
import isProduction from "new/local/utils/isProduction"
import App from "new/local/app/ui/App"
import Examples from "new/local/app/ui/Routes/Examples"
import OldExamples from "new/local/app/ui/Routes/OldExamples"
export default function Routes () {
  return (
    <BrowserRouter>
      {
        isProduction() ? <Route component={App} /> :
        (
          <Switch>
            <Route path="/examples" component={Examples} />
            <Route path="/old-examples" component={OldExamples} />
            <Route component={App} />
          </Switch>
        )
      }
    </BrowserRouter>
  )
}