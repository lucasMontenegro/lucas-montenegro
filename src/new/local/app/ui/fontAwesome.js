const { library } = require("@fortawesome/fontawesome-svg-core")
const { fab } = require("@fortawesome/free-brands-svg-icons")
const { fas } = require("@fortawesome/free-solid-svg-icons")
const { far } = require("@fortawesome/free-regular-svg-icons")
library.add(fab, fas, far)