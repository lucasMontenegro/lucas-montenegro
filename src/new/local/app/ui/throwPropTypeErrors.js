const { default: globals } = require("new/local/utils/globals")
const { default: isProduction } = require("new/local/utils/isProduction")
exports.default = function throwPropTypeErrors () {
  if (isProduction()) {
    return
  }
  const { error } = globals.console
  globals.console.error = function logErrorOrThrow (msg) {
    if (/invalid|failed/gi.test(msg) && /prop.?type/gi.test(msg)) {
      throw new Error(msg)
    }
    error.apply(this, arguments)
  }
}