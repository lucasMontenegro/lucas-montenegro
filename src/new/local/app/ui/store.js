import React from "react"
import { createStore, combineReducers } from "redux"
import { Provider } from "react-redux"
import Routes from "new/local/app/ui/Routes"
export const reducers = {
  core (state={}, action) {
    return state
  },
}
export default (
  <Provider store={createStore(combineReducers(reducers))}>
    <Routes />
  </Provider>
)