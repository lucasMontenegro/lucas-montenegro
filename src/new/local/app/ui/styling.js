const React = require("react")
const { default: CssBaseline } = require("@material-ui/core/CssBaseline")
const { ThemeProvider } = require("@material-ui/styles")
const { createMuiTheme } = require("@material-ui/core/styles")
require("typeface-roboto")
require("new/local/app/ui/fontAwesome")
const { default: store } = require("new/local/app/ui/store")
exports.default = (
  <ThemeProvider theme={createMuiTheme()}>
    <CssBaseline />
    {store}
  </ThemeProvider>
)