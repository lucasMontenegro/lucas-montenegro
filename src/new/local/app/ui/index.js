import ReactDOM from "react-dom"
import styling from "new/local/app/ui/styling"
import throwPropTypeErrors from "new/local/app/ui/throwPropTypeErrors"
import globals from "new/local/utils/globals"
throwPropTypeErrors()
ReactDOM.render(styling, globals.document.getElementById(`root`))