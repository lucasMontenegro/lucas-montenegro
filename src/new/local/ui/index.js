import ReactDOM from "react-dom"
import styling from "new/local/ui/styling"
import throwPropTypeErrors from "new/local/ui/throwPropTypeErrors"
import globals from "new/local/utils/globals"
throwPropTypeErrors()
ReactDOM.render(styling, globals.document.getElementById(`root`))