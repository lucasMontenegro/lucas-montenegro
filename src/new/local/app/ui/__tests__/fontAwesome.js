describe(`new/local/app/ui/fontAwesome`, () => {
  it(`should utilize the dependency APIs correctly`, () => {
    expect(jestUtils.getDependencies([
      `@fortawesome/fontawesome-svg-core`,
      `@fortawesome/free-brands-svg-icons`,
      `@fortawesome/free-solid-svg-icons`,
      `@fortawesome/free-regular-svg-icons`,
    ])).toMatchSnapshot()
  })
  it(`should run`, () => {
    require("new/local/app/ui/fontAwesome")
  })
})