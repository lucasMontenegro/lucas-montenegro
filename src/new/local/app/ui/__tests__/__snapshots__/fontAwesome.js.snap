// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/local/app/ui/fontAwesome should utilize the dependency APIs correctly 1`] = `
Object {
  "@fortawesome/fontawesome-svg-core": "^1.2.25",
  "@fortawesome/free-brands-svg-icons": "^5.11.2",
  "@fortawesome/free-regular-svg-icons": "^5.11.2",
  "@fortawesome/free-solid-svg-icons": "^5.11.2",
}
`;
